packages/x-date-pickers/src/internals/utils/date-utils.ts

```diff
@@ -124,6 +124,21 @@ export const mergeDateAndTime = <TDate>(
   return mergedDate;
 };
 
+export const isEqualTime = <TDate>(
```

Comment:
I am afraid this is risky if we support ms one day.
Can the ms be different? If not we could check them as well